portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,19 +33,41 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-    private List<String> messages;
-
-  @Override
-  public void init() {
-    messages = new ArrayList<>(Arrays.asList("Hi there!", "Welcome to my protfolio", "Have a great day :)"));
-  }
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Send the JSON as the response
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<String> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String comment = (String) entity.getProperty("comment");
+      comments.add(comment);
+    }
+
     response.setContentType("application/json;");
-    response.getWriter().println(new Gson().toJson(messages)),
+    Gson gson = new Gson();
```

Comment:
Can do this in a one-liner with the next line.

Reply:
regarding the toJson method- it is not static and therefore the gson object needs to be initialized 
https://www.javadoc.io/doc/com.google.code.gson/gson/2.8.5/com/google/gson/Gson.html#toJson-java.lang.Object-